### PR TITLE
RD-381 Upgrade Sequelize to version 6 and refactor DB Initializer

### PR DESCRIPTION
**Pull request by qooban**
## Description
This PR upgrades Sequelize to version 6. As `sequelize.import` method has been removed, DB Initializer had to be updated not to get a directory containing models and potential excludes, but already imported model files (by model files I understand files exporing a function with such signature: `function(sequelize: Sequelize, dataTypes: Sequelize.DataTypes): Sequelize.Model`).

Once this PR is merged I'm planning to release new major version: `5.0.0`.
Then the following PRs can be updated:
1. Stage - https://github.com/cloudify-cosmo/cloudify-stage/pull/1623
2. Composer - https://github.com/cloudify-cosmo/cloudify-blueprint-composer/pull/929

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests
1. Unit tests updated and passing.
2. Stage - tested internally, seems working good, but I didn't yet run all system tests.
3. Composer - tested internally, seems working good, only DB initializer changes required, not yet system tested.

Will run Stage and Composer system tests once we have new `cloudify-ui-common` version released.

## Documentation
N/A

### CODE REVIEW

File: backend/db.js [hunk 1]
```diff
@@ -41,22 +40,16 @@ function addHooks(sequelize, restart) {
  * @param {string | Array} dbConfig.url DB connection URL or an array of URLs
  * @param {Object} dbConfig.options DB connection options
  * @param {Object} loggerFactory object containing `getLogger` function
- * @param {string} modelsDir directory for models lookup
- * @param {Array} modelExcludes list of files to be excluded from model definitions discovered by reading models
- * directory as specified by `modelsDir` argument
+ * @param {(function(Sequelize, Sequelize.DataTypes): Sequelize.Model)[]} models array of functions returning sequelize model
```
Comment (qooban) [inline backend/db.js:43]: I'm open for name suggestions for this parameter, if you think it is not accurate.

Comment (kubama) [reply backend/db.js:43]: I guess it's a factory function pattern, so maybe `modelFactories`?

**Review by kubama — APPROVED** [review]

File: backend/db.js (hunk 1 above)
Comment (qooban) [reply backend/db.js]: Yes, make sense. Updated.

**Review by kubama — DISMISSED** [review]

**Review by kubama — APPROVED** [review]

**Comment by qooban** [comment]
@kubama When working on RD-1290 (migration of backend directory to TS) I found that the method used to get model name is not the best - we are accessing not "officialy" exposed `name` property. 
I see no method to access model name in Sequelize API. There's only [getTableName](https://sequelize.org/master/class/lib/model.js~Model.html#static-method-getTableName) to get DB table name, which can be different from model name.

So, to create better solution at once in this PR, I am think of having different structure for `modelFactories`, instead of having:
```((Sequelize, Sequelize.DataTypes) => Sequelize.Model)[]``` 
I'm thinking of 
```Record<string; (Sequelize, Sequelize.DataTypes) => Sequelize.Model>``` 
where the key will be the name of the model. 

That will require to pass something like (in Stage and Composer):
```
import ApplicationsModel from './models/ApplicationsModel';

const dbInitializer = new DbInitializer(getConfig().app.db, loggerFactory, { 'Applications': ApplicationsModel })
```
Possibly the best would be then if 'Applications' string was moved to a const in './models/ApplicationsModel' and used there to pass to `sequelize.define`.

WDYT?

**Comment by kubama** [comment]
@qooban I would leave the `model.name` approach. Being unable to access model name through the API seems like a design flaw in sequelize API.

**Comment by qooban** [comment]
> I would leave the model.name approach. Being unable to access model name through the API seems like a design flaw in sequelize API.

@kubama I was leaning towards the same decision. OK, I'm leaving it as is. No more changes planned then, but not to pollute `master` branch with breaking change for a week (https://github.com/cloudify-cosmo/cloudify-stage/pull/1623 and https://github.com/cloudify-cosmo/cloudify-blueprint-composer/pull/929 won't be merged today), I'm leaving this PR open till 4th October. I'll merge and release new version when I'm back.

**Pull request merged**